test/jasmine/tests/indicator_test.js

```diff
@@ -0,0 +1,438 @@
+            .then(done);
+        });
+
+        it('position it left of bullet', function(done) {
+            Plotly.newPlot(gd, [{
+                type: 'indicator',
+                value: 1,
+                title: {text: 'Value'},
+                mode: 'gauge',
+                gauge: {shape: 'bullet'}
+            }])
+            .then(function() {
+                gd.style.display = 'block';
+                var t = d3.selectAll('text.title').node();
+                var titleBBox = t.getBoundingClientRect();
+
+                var ax = d3.selectAll('g.bulletaxis').node();
+                var axBBox = ax.getBoundingClientRect();
+                expect(titleBBox.right < axBBox.left).toBe(true);
+            })
+            .catch(failTest)
+            .then(done);
+        });
+    });
+
+    it('restyle between modes', function(done) {
+        function assertElementCnt(sel, cnt) {
+            var el = d3.selectAll(sel);
+            expect(el.size()).toBe(cnt, 'selection "' + sel + '" does not have size ' + cnt);
+        }
+        function assertGauge(shape, cnt) {
+            assertElementCnt('g.' + shape, cnt);
+            assertElementCnt('g.' + shape + 'axis', cnt);
+        }
+        function assert(flags) {
+            // flags is an array denoting whether the figure [hasNumber, hasDelta, hasAngular, hasBullet]
+            assertElementCnt('tspan.number', flags[0]);
+            assertElementCnt('tspan.delta', flags[1]);
+            assertGauge('angular', flags[2]);
+            assertGauge('bullet', flags[3]);
+        }
+
+        Plotly.newPlot(gd, [{
+            type: 'indicator',
+            value: 100,
+            mode: 'number+delta+gauge'
+        }])
+        .then(function() {
+            assert([1, 1, 1, 0]);
+            return Plotly.restyle(gd, 'mode', 'number+delta');
+        })
+        .then(function() {
+            assert([1, 1, 0, 0]);
+            return Plotly.restyle(gd, 'mode', 'number');
+        })
+        .then(function() {
+            assert([1, 0, 0, 0]);
+            return Plotly.restyle(gd, 'mode', 'delta');
+        })
+        .then(function() {
+            assert([0, 1, 0, 0]);
+            return Plotly.restyle(gd, 'mode', 'gauge');
+        })
+        .then(function() {
+            assert([0, 0, 1, 0]);
+            return Plotly.restyle(gd, 'gauge.shape', 'bullet');
+        })
+        .then(function() {
+            assert([0, 0, 0, 1]);
+            return Plotly.restyle(gd, 'mode', 'number+delta+gauge');
+        })
+        .then(function() {
+            assert([1, 1, 0, 1]);
+        })
+        .catch(failTest)
+        .then(done);
+    });
+});
+
+describe('Indicator animations', function() {
+    var gd;
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+    afterEach(destroyGraphDiv);
+
+    it('should be able to transition via `Plotly.react`', function(done) {
+        var mock = {data: [{type: 'indicator', value: 100}], layout: {}};
```

Comment:
Can we also check that `indicator` traces with bullet/angular gauges can also be smoothly transitioned using `Plotly.react`?

Reply:
Done in e6d8525ab07c56ca10bdbac6762b06d1d57bf830